grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/rule/InferenceRule.java

```diff
@@ -151,37 +151,47 @@ public Atom getRuleConclusionAtom() {
     }
 
     /**
-     * @param parentAtom from which constrained are propagated
-     * @return the inference rule with constraints
+     * @param parentAtom atom containing constraints (parent)
+     * @param u rule unifier
+     * @param pu permutation unifier
+     * @return rule with propagated constraints from parent
      */
-    public  InferenceRule propagateConstraints(Atom parentAtom){
+    public InferenceRule propagateConstraints(Atom parentAtom, Unifier u, Unifier pu){
```

Comment:
Avoid symbolic programming where please. To an outsider `p` and `pu`don't provide any context. I know you clarify with the javadocs but often the code gets updated without the javadocs being updated so things become even more confusing.